### ISSUE TITLE
[Accessibility] : AlertDialog semanticLabel not read in iOS 18

### ISSUE THREAD

**Issue by Zigotote** [flutter/flutter#161081]
### Steps to reproduce

The semanticLabel of my AlertDialog is not read by VoiceOver in iOS 18. It's working well on iOS 17 and above, and on Android.
I'm facing the issue with Flutter 3.24 and 3.27

### Expected results

semanticLabel should be read when AlertDialog is focused

### Actual results

semanticLabel is not read when AlertDialog is focused

### Code sample

<details open><summary>Code sample</summary>

```dart
showDialog(
    useRootNavigator: true,
    barrierDismissible: false,
    context: context,
    builder: (dialogContext) {
      return AlertDialog(
        semanticLabel: "Alert !",
        title: Text("My tittle"),
        content: Text("My content"),
        actions: [TextButton(label: Text("OK"), onPressed: context.pop))]
        scrollable: true,
      );
    },
  );
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.27.1, on macOS 15.1 24B83 darwin-x64, locale fr-FR)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 16.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.3)
[✓] IntelliJ IDEA Ultimate Edition (version 2023.2.4)
[✓] Connected device (2 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
@Zigotote 
Can you check if there's any setting enabled on your phone which might be preventing this ?
Something like `Settings -> Accessibility -> Voice Over -> VoiceOver Recognition -> Screen Recognition -> Apply to Apps` if your app is enabled for it ? and if yes by chance, disabling it to check if it works or not ?

**Comment by Zigotote** [comment]
Hi @darshankawar I tried to disable and enable this settings.
When it's disabled, the AlertDialog has the focus when it's opened but the text "Alert !" isn't read (as described in my issue).
When it's enabled the behavior is quite strange (the button which opens the AlertDialog stays focused when the AlertDialog is displayed... and "Alert !" is still not read)

**Comment by darshankawar** [comment]
Thanks for the update. I verified on iOS 18.1 device and was able to replicate the reported behavior.

```
stable : 3.27.1
master : 3.28.0-2.0.pre.38699
```

**Comment by chunhtai** [comment]
the label will be used to send UIAccessibilityScreenChangedNotification in iOS https://github.com/flutter/flutter/blob/5a3541c40c08c3c978e0bde4f528b0ae52c38e04/engine/src/flutter/shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm#L214

We need to investigate why it doesn't work in iOS.

@Zigotote Does this only apply to alertdialog? do you see the same issue in other types of dialog?

**Comment by Zigotote** [comment]
@chunhtai  I only use AlertDialog in my app so I can't tell if it happens on other types of dialogs unfortunately...

**Comment by hannah-hyj** [comment]
@chunhtai  I tested SimpleDialog and it has the same issue, the route change is not announced.

**Comment by chunhtai** [comment]
looking at the doc https://developer.apple.com/documentation/uikit/uiaccessibility/notification/screenchanged?language=objc, it no longer mention providing a string to announce, but only the element. We could workaround this by issuing a announce notification instead

**Comment by vashworth** [comment]
cc @LouiseHsu are you interested in this?

**Comment by vashworth** [comment]
Since @chunhtai said this seems severe and can be reproduced on multiple dialogs, marking as P1

**Comment by Zigotote** [comment]
That's strange because, for what I know, for an app to be accessible all elements that appear on the screen should be announced. As said in https://www.wcag.com/developers/4-1-3-status-messages/
I don't understand why iOS would remove this feature... 
Anyway, thanks for investigating !
